data/integration_tests/percolator/IntegrationTest_Percolator_SpeedGraph.py

```diff
@@ -0,0 +1,165 @@
+    return parser.parse_args()
+
+def makeReducedFile(fileNameOriginal, fileNameReduced, numLines):
+    fileReduced = open(fileNameReduced, "w")
+    counter = 0
+    with open(fileNameOriginal) as f:
+        for line in f:
+            counter += 1
+            if (counter > numLines):
+                break
+            fileReduced.write(line)
+    fileReduced.close()
+
+from itertools import (takewhile,repeat)
+def rawincount(filename): ####https://stackoverflow.com/a/27518377/7202012
+    f = open(filename, 'rb')
+    bufgen = takewhile(lambda x: x, (f.raw.read(1024*1024) for _ in repeat(None)))
+    return sum( buf.count(b'\n') for buf in bufgen )
+
+def getTemporaryInDataFileName(number):
+    return "psmInData" + str(number) + ".tab"
+
+def getTerminalCommand(args, data):
+    result = args.python_version
+    result += " " + testScriptPath
+    result += " -d " + data
+    if len(args.flags) > 0:
+        result += " -f \"" + args.flags + "\""
+    result += " -r " + str(args.runs)
+    if args.await_tests is True:
+        result += " -a "
+    print("RES: " + result)
+    return result
+    
+def addElement(dict, key, value):
+    if key not in dict:
+        dict[key] = []
+    dict[key].append(value)
+
+def getChartLinesShortened(tmpNumLines):
+    tmpNumLines /= 1000000
+    return int(tmpNumLines)
+
+def getBarChart(y_valuesWall, y_valuesCPU, x_numLines):
+    d1 = {}
+    d2 = {}
+    data = {}
+    for i in range (0,len(x_numLines)):
+        d1[x_numLines[i]] = y_valuesWall[i]
+        d2[x_numLines[i]] = y_valuesCPU[i]
+    data['Wall'] = d1
+    data['CPU'] = d2
+    df = pd.DataFrame(data)
+    df.plot(kind='bar', width=0.4, color=['darkgray','gray'])
+    colors = {'Wall':'darkgray', 'CPU':'gray'}         
+    labels = list(colors.keys())
+    handles = [plt.Rectangle((0,0),1,1, color=colors[label]) for label in labels]
+    plt.legend(handles, labels)
+    plt.tight_layout()
+    plt.setp(plt.xticks()[1], rotation=0)
+    return plt
+
+
+args = getArguments()
+testDataFilePath = args.data
+
+#################
+numLines = 5000000
+numDataFiles = 0
+numLinesMax = rawincount(testDataFilePath)
+for i in range(0, int(numLinesMax/numLines)):
+    numDataFiles += 1
+    makeReducedFile(testDataFilePath, testDataFolderPath + "/" + getTemporaryInDataFileName(i+1), numLines*numDataFiles)
+#################
+runTimes = []
+for i in range(1, numDataFiles+1):
+    terminalCommand = getTerminalCommand(args, testDataFolderPath + "/" + getTemporaryInDataFileName(i))
+    p = subprocess.Popen(terminalCommand, stderr=PIPE , stdout=PIPE , text=True, shell=True)
+    print("Running subprocess: " + str(p.pid))
+    p.wait()
+    result = p.stdout.read()
+    print(p.stderr.read())
+    result = result.splitlines()
+    for item in result:
+        if "Min" in item:
+            runTime = re.findall(r"[-+]?\d*\.\d+|\d+", item)
+            runTime = [float(i) for i in runTime]
+            runTimes.append(runTime)
```

Comment:
This substring/regex setup feels like it could easily malfunction if e.g. someone added a timestamp to the program output lines. At the least, there should be a comment in this file noting this dependency.